source/Engine.cpp

```diff
@@ -837,8 +837,19 @@ void Engine::Step(bool isActive)
 		targetSwizzle = target->GetSwizzle();
 		info.SetString("mission target", target->GetPersonality().IsTarget() ? "(mission target)" : "");
 
+		// Only update the "active" state shown for the target if it is
+		// in the current system and targetable, or owned by the player.
 		int targetType = RadarType(*target, step);
-		info.SetOutlineColor(GetTargetOutlineColor(targetType));
+		const bool blinking = targetType == Radar::BLINK;
+		if(!blinking && ((target->GetSystem() == player.GetSystem() && target->IsTargetable()) || target->IsYours()))
+			targetWasInactive = targetType == Radar::INACTIVE;
+		if(blinking)
+			info.SetOutlineColor(GetTargetOutlineColor(Radar::BLINK));
+		else if(targetWasInactive)
+			info.SetOutlineColor(GetTargetOutlineColor(Radar::INACTIVE));
+		else
+			info.SetOutlineColor(GetTargetOutlineColor(targetType));
```

Comment:
I think we can make this simpler by just caching the targetType (and only updating it when the target is visible), note that this suggestion does require the variable in the header file to change to `lastTargetType`.

```suggestion
		const bool blinking = targetType == Radar::BLINK;
		if(blinking || ((target->GetSystem() == player.GetSystem() && target->IsTargetable()) || target->IsYours()))
			lastTargetType = targetType;
		info.SetOutlineColor(GetTargetOutlineColor(targetType));
```

Reply:
You're right, we don't want to cache blinking. But should the condition then be `if(!blinking &&` instead of `if(!blinking ||` to avoid such caching?

I also don't like the look of the ternary, the other option looks more clear.

Reply:
I think this last remark (if it should be `||` or `&&` on line 844) is still valid.

Rest of the PR looks good to me.

Reply:
You're right. I don't know why I changed it.